clang-cl error in 'platform/cert_capi.c'

I've reported on clang-cl compile errors before.

Again, there is this compile error:
```
clang-cl -c @clang-cl.args -Fo./clang_obj/platform_cert_capi.obj platform/cert_capi.c
platform/cert_capi.c(566,1): error: conflicting types for 'CxPlatCertSelect'
CxPlatCertSelect(
^
./inc\quic_cert.h(44,1): note: previous declaration is here
CxPlatCertSelect(
^
1 error generated.
```

The prototype and implementation are different AFAICS.
Preprocessed and Astyle'd output of `inc/quic_cert.h`:
```c
BOOLEAN CxPlatCertSelect (QUIC_CERTIFICATE *Certificate,
  const uint16_t *SignatureAlgorithms,
  size_t SignatureAlgorithmsLength,
  uint16_t *SelectedSignature);
```

And of `platform/cert_capi.c`:
```c
BOOLEAN CxPlatCertSelect (PCCERT_CONTEXT CertCtx,
  const UINT16 *SignatureAlgorithms,
  size_t SignatureAlgorithmsLength,
  UINT16 *SelectedSignature)
{
  if (CertCtx == ( (void *) 0))
  {
    *SelectedSignature = SignatureAlgorithms[0];
    return 1;
  }
  ....
```

A difference in 1st argument.
I'm frankly amazed that MSVC-2019 is allowing this to compile.

Update:
And using MinGW (TDM-gcc v9.2.0) with `"openssl"`, I get the same compile error:
```
platform/cert_capi.c:566:1: error: conflicting types for 'CxPlatCertSelect'
  566 | CxPlatCertSelect(
      | ^~~~~~~~~~~~~~~~
In file included from platform/platform_internal.h:16,
                 from platform/cert_capi.c:17:
./inc/quic_cert.h:44:1: note: previous declaration of 'CxPlatCertSelect' was here
   44 | CxPlatCertSelect(
      | ^~~~~~~~~~~~~~~~
```

--- a/src/platform/cert_capi.c
+++ b/src/platform/cert_capi.c
@@ -564,7 +564,7 @@ CxPlatCertFree(
 _Success_(return != FALSE)
 BOOLEAN
 CxPlatCertSelect(
-    _In_opt_ PCCERT_CONTEXT CertCtx,
+    _In_opt_ QUIC_CERTIFICATE* Certificate,
     _In_reads_(SignatureAlgorithmsLength)
         const uint16_t *SignatureAlgorithms,
     _In_ size_t SignatureAlgorithmsLength,
@@ -577,6 +577,8 @@ CxPlatCertSelect(
     // Low byte of SignatureAlgorithms[] is the TLS SignatureAlgorithm:
     //  anonymous(0), rsa(1), dsa(2), ecdsa(3)
     //
+    
+    PCCERT_CONTEXT CertCtx = (PCCERT_CONTEXT)Certificate;
 
     if (CertCtx == NULL) {
         *SelectedSignature = SignatureAlgorithms[0];

